web/src/components/table/data-table-column-visibility-filter.tsx

```diff
@@ -10,22 +10,28 @@ import {
   DropdownMenuContent,
   DropdownMenuTrigger,
   DropdownMenuCheckboxItem,
+  DropdownMenuSeparator,
+  DropdownMenuLabel,
 } from "@/src/components/ui/dropdown-menu";
 import { type VisibilityState } from "@tanstack/react-table";
 import { ChevronDown, Columns } from "lucide-react";
 import { type LangfuseColumnDef } from "@/src/components/table/types";
 import { usePostHogClientCapture } from "@/src/features/posthog-analytics/usePostHogClientCapture";
 
 interface DataTableColumnVisibilityFilterProps<TData, TValue> {
-  columns: LangfuseColumnDef<TData, TValue>[];
+  baseColumns: LangfuseColumnDef<TData, TValue>[];
```

Comment:
is there a native way to do this with tanstack table, e.g. grouping columns?

https://tanstack.com/table/v8/docs/guide/column-defs#column-helpers

Reply:
Yes, we could use groupings (also shown in table with additional group header) in the future, and infer groups of columns based on this 👍 I quite like the approach. 

However, this requires a bit of a refactor of our table component and types that I'd prefer to decouple from this functionality PR. I'd prefer to have a separate PR in the near future to add grouping support to our table component, wdyt?